### PR TITLE
feat: Spell 23-04-2026 (DEV-1593)

### PR DESCRIPTION
**Pull request by supercontracts**
# 2026-04-23 Spark Spell

## Forum Post
https://forum.skyeco.com/t/april-23-2026-proposed-changes-to-spark-for-upcoming-spell/27831

## Dependencies on Sky Core spell
None

## Notes for Reviewers
None

## Conditions for Spell Deployment
- [ ] All spell-specific tests are passing.
- [ ] All E2E tests are passing.
- [ ] All newly deployed addresses have been reviewed and added to the spark-address-registry in a PR that has been merged.
- [ ] All newly deployed addresses have been added to the Immunefi bug bounty program.
- [ ] There is complete test coverage of all changes in the spell, including any relevant edge cases.
- [ ] At least three approvals have been given from reviewers.
- [ ] CODE_RABBIT.md comment has been posted, AI review performed.
- [ ] Octane review performed.

## Conditions for Spell Handoff to Sky
- [ ] All spells payloads have been deployed and verified on Etherscan.
- [ ] All spell-specific tests are passing (including spell bytecode verification).
- [ ] All E2E tests are passing.
- [ ] Spell caster is passing in CI.
- [ ] At least three approvals have been given from reviewers on deployed spell.
- [ ] Comment from at least three reviewers confirming address of mainnet spell payload.

## Conditions for Spell Merge
- [ ] Spell handoff performed to Sky team on Discord.
- [ ] Spell handoff performed to Sky team on Signal.
- [ ] Spell address confirmed to be correct in sky-ecosystem/spells-mainnet repo open PR through comments by at least three reviewers.
- [ ] Sky deployed spell confirmed through a comment in this PR to contain correct Spark payload address by at least three reviewers.

## Summary by CodeRabbit

* **New Features**
  * Added governance proposal with interest rate parameter updates for WETH and USDT, rETH reserve freeze/deprecation, and USDS grant distributions to foundation accounts.

* **Tests**
  * Added comprehensive test suite validating interest rate updates, rETH freeze functionality, and treasury transfer execution.

* **Documentation**
  * Added detailed report documenting all reserve parameter changes for the proposal.

### CODE REVIEW

File: src/proposals/20260423/Spell_20260423.t.sol [hunk 1]
```diff
@@ -0,0 +1,172 @@
...
+        IERC20(Ethereum.RETH).approve(address(pool), type(uint256).max);
+
+        // User can't supply.
+        vm.expectRevert(bytes("28"));  // RESERVE_FROZEN
+        pool.supply(Ethereum.RETH, reserveAmount, testUser, 0);
+
+        // User can't borrow.
+        vm.expectRevert(bytes("28"));  // RESERVE_FROZEN
+        pool.borrow(Ethereum.RETH, debtAmount, 2, 0, testUser);
+    }
```
Comment (certora-spark-review) [inline src/proposals/20260423/Spell_20260423.t.sol:201]: The current test only verifies that no new supply or borrow actions are possible. Please consider adding additional tests to ensure that existing users are not negatively affected after execution of the Spell.

We can use this test as a reference: [_testUserActionsOnFrozenReserve](https://github.com/sparkdotfi/spark-spells/pull/150/changes#diff-e60acae25567a368ee6f1fa368e864dea3cb19ef85551c086ef397890f9d0f1c:~:text=function-,_testUserActionsOnFrozenReserve,-(address%20reserveAsset)

Comment (supercontracts) [reply src/proposals/20260423/Spell_20260423.t.sol:201]: added

**Comment by lucas-manuel** [comment]
@coderabbitai review

Please use the following master prompt for your analysis:

### CodeRabbit AI Master Prompt for Spark Spell Audits

**Objective:** To perform an expert-level, security-first code review of a Spark governance spell, ensuring it perfectly and safely implements the specification outlined in a corresponding forum post.

**Persona:** You are CodeRabbit AI, but for this review, you will adopt the persona of an expert-level smart contract auditor and tech lead with a security-first mindset. Your standards for correctness, security, and clarity are exceptionally high. The code you are reviewing is for a high-stakes DeFi protocol where any error can lead to significant financial loss. Be meticulous, rigorous, and leave no stone unturned.

---

**[CONTEXT & INSTRUCTIONS]**

Your task is to conduct a comprehensive audit of a new Spark Spell. Here are your source materials:

*   **The Specification (Source of Truth):** 

```
Summary
[Ethereum] SparkLend - Update ETH Interest Rate Model
[Ethereum] SparkLend - Update USDT Interest Rate Model
[Ethereum] SparkLend - Deprecate rETH
Recurring Items

[Ethereum] SparkLend - Claim SparkLend Reserves (Exec)
[Ethereum] Spark Treasury - Monthly Grants for Spark Foundation and Spark Assets Foundation
Rationale
[Ethereum] SparkLend - Update ETH Interest Rate Model
Recently we have observed suboptimal market dynamics in the SparkLend ETH market, with utilization typically at or just below the optimal ratio (indicating that equilibrium rates may exceed the optimal borrow rate), while occasional shifts above the optimal utilization ratio cause significant rate spikes that impact borrower UX. Increasing slope1 while reducing slope2 should result in increased yield for ETH suppliers and protocol revenue, while at the same time improving the borrower experience and predictability of looping strategy returns.

Change summary:

ETH interest rate model
slope1: Lido stETH yield - 0.1%
slope2: 75%
[Ethereum] SparkLend - Update USDT Interest Rate Model
It has been some time since the SparkLend USDT rate model was updated, and it is currently above market, resulting in low utilization ratios and lower market efficiency. Reducing the slope1 spread from 1.25% to 0.5% will allow the market to reach equilibrium at a higher utilization ratio, which should improve cost effectiveness for borrowers at any given target supply rate implemented by the Spark Liquidity Layer. This will also free up additional USDT liquidity that Spark can allocate elsewhere to generate revenue.

Change summary:

USDT interest rate model
slope1: SSR + 0.5% (currently +1.25%)
[Ethereum] SparkLend - Deprecate rETH
rETH has seen low and declining usage on SparkLend, and no longer generates sufficient protocol revenue to justify the incremental tail risk exposure introduced into the SparkLend market. We plan to deprecate rETH, with a first stage freezing new supply and borrowing of rETH and reducing LTV to 0% to limit new collateral usage among existing suppliers. Because the liquidation threshold is not being changed, there is no impact on existing user positions for the time being.

Change summary:

Freeze rETH supply and borrow
Set rETH LTV to 0%
[Ethereum] SparkLend - Claim SparkLend Reserves (Exec)
This is a recurring item that can go directly to executive vote based on approval in the Atlas. Claiming and consolidating SparkLend reserves increases Spark’s available risk capital, allowing for more revenue generating capacity for the SLL.

Parameter summary:

Claim all reserves
Transfer USD stablecoin reserves to ALM Proxy
Transfer non-USD stablecoin reserves to Spark Operations Multisig 0x2E1b01adABB8D4981863394bEa23a1263CBaeDfC to be liquidated
[Ethereum] Spark Treasury - Monthly Grants for Spark Foundation and Spark Assets Foundation
Per approval in the Spark Artifact and the previous governance proposal authorizing payments for Q2 2026, this spell will transfer the below amounts to Spark Foundation and Spark Assets Foundation.

Change summary:

Spark Foundation Grant
Recipient address: 0x92e4629a4510AF5819d7D1601464C233599fF5ec
Transfer amount: 1,100,000 USDS
Spark Asset Foundation Grant
Recipient address: 0xEabCb8C0346Ac072437362f1692706BA5768A911
Transfer amount: 100,000 USDS
```

Your primary mission is to ensure the smart contract code within the pull request **perfectly, correctly, and safely** implements the changes described in the forum post. Any deviation, ambiguity, or potential risk must be flagged.

Follow this comprehensive review checklist.

---

#### **Part 1: Specification-to-Code Verification**

This is the most critical step. The code must be a perfect translation of the forum post's intent.

1.  **Parameter & Constant Cross-Reference:**
    *   Scrutinize every numerical value, address, and constant defined in the `Spell.sol` file.
    *   Compare each one directly against the values specified in the forum post.
    *   Create a markdown table in your review that explicitly maps each parameter change from the post to the specific line of code that implements it.
    *   **Flag any mismatch, however small (e.g., a single digit off, a potential precision error, a typo in an address).**

2.  **Logic & Function Call Validation:**
    *   Verify that the functions being called on target contracts (e.g., `POOL_CONFIGURATOR`, `ACL_MANAGER`) are the correct ones to achieve the stated goals in the forum post.
    *   Confirm that the `target` contract addresses hardcoded in the spell are correct for the intended network.

#### **Part 2: Security & Risk Analysis**

Adopt a hacker's mindset. How could this spell be exploited or fail?

1.  **Access Control:** Ensure the spell only calls functions that the `SparkSpellExecutor` has the authority to call.
2.  **Execution Atomicity:** Confirm the entire spell executes as a single, atomic transaction. Are there any failure modes that could leave the protocol in an inconsistent or vulnerable state?
3.  **External Call Safety:** Review all external calls. Are they to trusted, audited contracts? Is there any risk of re-entrancy, even if unlikely in a governance spell?
4.  **Input Validation:** While most spell inputs are hardcoded, double-check that no inputs could inadvertently cause harm (e.g., setting a threshold to zero, setting LTV higher than the liquidation threshold).

#### **Part 3: Test Coverage & Validation**

Do not just check if the tests pass. Critically evaluate the quality and thoroughness of the test suite (`Spell.t.sol`).

1.  **State Verification:** The tests **must** prove that the post-execution state of the protocol matches the specification. For every parameter change in the forum post, there must be a corresponding `assertEq` in the test file that confirms the new value is set correctly after the spell is executed.
2.  **Forking Sanity Check:** The tests run on a mainnet fork. Verify that the forking block number is recent and relevant.
3.  **Completeness:** Do the tests cover *all* changes proposed in the forum post? Identify any proposed change that is not explicitly tested.
9:14
4.  **Edge Cases:** While less common for simple parameter updates, ask yourself: are there any implicit assumptions or edge cases the tests are not considering?

#### **Part 4: Code Quality & Repository Conventions**

1.  **Clarity & Readability:** Is the code clean, well-commented, and easy to understand?
2.  **Natspec Comments:** Ensure all functions and parameters have clear, compliant Natspec documentation.
3.  **Conventions:** Does the spell follow the established structure and naming conventions of the `spark-spells` repository?

---

**[OUTPUT FORMAT]**

Structure your review for maximum clarity:

1.  **Summary:** Start with a brief, high-level summary of your findings. State clearly whether you approve the changes, approve with comments, or request changes.
2.  **Specification-to-Code Mapping:** Include the markdown table you created in Part 1.
3.  **Findings:** Group your findings by severity:
    *   **🔴 Critical:** Issues that could lead to direct financial loss, security breaches, or major protocol malfunctions.
    *   **🟠 High:** Issues that could lead to unexpected behavior or undermine the spell's intent.
    *   **🟡 Medium:** Gaps in test coverage, potential gas inefficiencies, or deviations from best practices.
    *   **🔵 Informational/Nitpicks:** Suggestions for improving code clarity, comments, or style.

For each finding, provide a clear description, the location in the code, its potential impact, and a concrete recommendation for remediation.

**Review by deluca-mike — DISMISSED** [review]

File: src/proposals/20260423/Spell_20260423.t.sol [hunk 2]
```diff
@@ -0,0 +1,253 @@
...
+
+        vm.prank(testUser);
+        reth.approve(address(pool), type(uint256).max);
+
+        vm.startPrank(testUser);
+
+        pool.supply(address(reth), 2 * rethAmount, testUser, 0);
+        pool.borrow(address(usdc), 2 * usdcAmount, 2,        0, testUser);
+
+        vm.stopPrank();
```
Comment (TheMj0ln1r) [inline src/proposals/20260423/Spell_20260423.t.sol]:
```suggestion
        vm.startPrank(testUser);

        reth.approve(address(pool), type(uint256).max);

        pool.supply(address(reth), 2 * rethAmount, testUser, 0);
        pool.borrow(address(usdc), 2 * usdcAmount, 2,        0, testUser);

        vm.stopPrank();
```

File: src/proposals/20260423/Spell_20260423.t.sol [hunk 3]
```diff
@@ -0,0 +1,253 @@
...
+
+        // Step 2: Execute spell
+
+        _executeAllPayloadsAndBridges();
+
+        // Step 3: Ensure user can't supply or borrow.
+
+        deal(address(reth), testUser, rethAmount);
+
```
Comment (TheMj0ln1r) [inline src/proposals/20260423/Spell_20260423.t.sol]: deal is not required

File: src/proposals/20260423/Spell_20260423.t.sol [hunk 4]
```diff
@@ -0,0 +1,253 @@
...
+        // User can't borrow.
+        vm.expectRevert(abi.encode("57"));  // As LTV is 0, user can't borrow.
+        pool.borrow(address(usdc), usdcAmount, 2, 0, testUser);
+
+        vm.stopPrank();
+
+        // Step 4: User can repay the debt.
+
+        deal(address(usdc), testUser, usdcAmount);
+
```
Comment (TheMj0ln1r) [inline src/proposals/20260423/Spell_20260423.t.sol]: deal usdc is not required, already got usdc from borrow

File: src/proposals/20260423/Spell_20260423.t.sol [hunk 5]
```diff
@@ -0,0 +1,253 @@
...
+        vm.stopPrank();
+
+        // Step 5: User can withdraw the collateral.
+
+        vm.startPrank(testUser);
+
+        pool.withdraw(address(reth), 1 * 10 ** IERC20Metadata(address(reth)).decimals(), testUser);
+
+        vm.stopPrank();
```
Comment (TheMj0ln1r) [inline src/proposals/20260423/Spell_20260423.t.sol:182]: Step 3 to 5 can be done in one vm.startPrank() block

File: src/proposals/20260423/Spell_20260423.t.sol [hunk 6]
```diff
@@ -0,0 +1,253 @@
...
+
+        assets[0]  = address(reth);
+        sources[0] = mockOracle;
+
+        vm.prank(Ethereum.SPARK_PROXY);
+        AaveOracle(SparkLend.AAVE_ORACLE).setAssetSources(assets, sources);
+
+        deal(address(usdc), testUser, usdcAmount);
+        deal(address(reth), testUser, rethAmount);
+
```
Comment (TheMj0ln1r) [inline src/proposals/20260423/Spell_20260423.t.sol]: reth balance is not required.

File: src/proposals/20260423/Spell_20260423.t.sol [hunk 7]
```diff
@@ -0,0 +1,253 @@
...
+        pool.repay(address(usdc), usdcAmount, 2, testUser);
+
+        vm.stopPrank();
+
+        // Step 5: User can withdraw the collateral.
+
+        vm.startPrank(testUser);
+
+        pool.withdraw(address(reth), 1 * 10 ** IERC20Metadata(address(reth)).decimals(), testUser);
+
```
Comment (TheMj0ln1r) [inline src/proposals/20260423/Spell_20260423.t.sol:181]: For more readability
```suggestion
        pool.withdraw(address(reth), rethAmount, testUser);

```

Comment (supercontracts) [reply src/proposals/20260423/Spell_20260423.t.sol:181]: don't want to withdraw `rethAmount` here as we want to liquidate in further steps.

Comment (TheMj0ln1r) [reply src/proposals/20260423/Spell_20260423.t.sol:181]: The liquidation can still happen after withdrawing rethAmount. But, we can leave this as is.

**Review by certora-spark-review — APPROVED** [review]
Approving the spell for deployment.

**Review by TheMj0ln1r — APPROVED** [review]
Approving the Spell deployment

- [x]  Verify the spell deployment : `0x160158d029697FEa486dF8968f3Be17a706dF0F0`
    
    ```solidity
    ➜  spark-spells git:(dev-1593-spell-23042026) ✗ forge verify-bytecode 0x160158d029697FEa486dF8968f3Be17a706dF0F0 SparkEthereum_20260423 
    
    Verifying bytecode for contract SparkEthereum_20260423 at address 0x160158d029697FEa486dF8968f3Be17a706dF0F0
    [⠒] Compiling...
    [⠒] Compiling 1 files with Solc 0.8.25
    [⠘] Solc 0.8.25 finished in 13.98s
    
    Creation code matched with status full
    Runtime code matched with status full
    ```

- [x]  Check forum post link - https://forum.skyeco.com/t/april-23-2026-proposed-changes-to-spark-for-upcoming-spell/27831#p-106130-ethereum-sparklend-update-eth-interest-rate-model-3
- [x]  Check description for all payloads
- [x]  Check spark-address-registry commit matches current HEAD
- SparkLend IRM update - ETH - [1] - DONE
    - [x]  Correct IRM variant identified (RateTargetKink vs RateTargetBase) - `RateTargetKink`
    - [x]  The new IRM should be the same type as Old IRM(unless the forum explicitly describes a migration)
    - [x]  setReserveInterestRateStrategyAddress called with correct asset address and new IRM address - WETH, NEW_WETH_IRM
    - [x]  New IRM contract address constant matches the deployed contract - `0xDFB6206FfC5BA5B48D2852370ee6A1bf6887476a`
    - [x]  Test uses correct helper (`_testRateTargetKinkIRMUpdate`)
    - [x]  Old IRM address in test matches the currently deployed IRM on-chain
    - [x]  New IRM address in test matches the constant in the payload
    - [x]  Old params in test match current on-chain state
    - [x]  New params in test match forum post values (spreads, slopes, optimal ratio)
        - [x]  slope1 : `-0.001e27`
        - [x]  slope2 : `0.75e27`
        - [x]  optimalUsageRation and BaseRate - unchanged
    - [x]  If multiple assets share the same new IRM address, confirm forum intends that - Only one asset per IRM
    - [x]  Every asset listed for IRM update in forum has both a payload action and a test
    - [x]  Verify the deployed IRM contract
        - [x]  Contract is verified on Etherscan
        - [x]  Source matches `lib/sparklend-advanced/src/RateTargetKinkInterestRateStrategy.sol`
        - [x]  Constructor args decoded from Etherscan match expected values:
            - [x]  `provider` , correct PoolAddressesProvider - `POOL_ADDRESSES_PROVIDER`
            - [x]  `rateSource` , correct rate oracle (same as old IRM's RATE_SOURCE)
            - [x]  `optimalUsageRatio` - `0.90e27` unchanged
            - [x]  `baseVariableBorrowRate` (Kink) - 0 - unchanged
            - [x]  `variableRateSlope1Spread` (Kink) -  `-0.001e27`
            - [x]  `variableRateSlope2`  - `0.75e27`
            - [x]  `_assertBytecodeMatches()` passes in the test (auto-verified by `_testRateTargetKinkIRMUpdate` helper)
    - [x]  Manual bytescode verification :
        
        ```solidity
        ➜  sparklend-advanced git:(master) POOL_ADDRESSES_PROVIDER="0x02C3eA4e34C0cBd694D2adFa2c690EECbC1793eE" RATE_SOURCE="0xaBc99f366D2bE1f4e5b8DFC0F561a751dd836246" OPTIMAL_USAGE_RATIO="900000000000000000000000000" BASE_BORROW_RATE="0" SLOPE_1="-1000000000000000000000000" SLOPE_2="750000000000000000000000000"
        
        ➜  sparklend-advanced git:(master) forge verify-bytecode 0xDFB6206FfC5BA5B48D2852370ee6A1bf6887476a RateTargetKinkInterestRateStrategy --encoded-constructor-args $(cast abi-encode "constructor(address,address,uint256,uint256,int256,uint256)" $POOL_ADDRESSES_PROVIDER $RATE_SOURCE $OPTIMAL_USAGE_RATIO $BASE_BORROW_RATE $SLOPE_1 $SLOPE_2)
        Verifying bytecode for contract RateTargetKinkInterestRateStrategy at address 0xDFB6206FfC5BA5B48D2852370ee6A1bf6887476a
        [⠊] Compiling...
        No files changed, compilation skipped
        Creation code matched with status full
        Runtime code matched with status full
        ```
        
- SparkLend IRM update - USDT - [2] - DONE
    - [x]  Correct IRM variant identified (RateTargetKink vs RateTargetBase) - `RateTargetKink`
    - [x]  The new IRM should be the same type as Old IRM(unless the forum explicitly describes a migration)
    - [x]  setReserveInterestRateStrategyAddress called with correct asset address and new IRM address - USDT, NEW_USDT_IRM
    - [x]  New IRM contract address constant matches the deployed contract - `0x4E494988E68e6Fc52309BE4937869e27F0C304AC`
    - [x]  Test uses correct helper (`_testRateTargetKinkIRMUpdate`)
    - [x]  Old IRM address in test matches the currently deployed IRM on-chain
    - [x]  New IRM address in test matches the constant in the payload
    - [x]  Old params in test match current on-chain state
    - [x]  New params in test match forum post values (spreads, slopes, optimal ratio)
        - [x]  slope1 : `0.005e27`
        - [x]  slope2 - unchanged
        - [x]  optimalUsageRation and BaseRate - unchanged
    - [x]  If multiple assets share the same new IRM address, confirm forum intends that - one asset per IRM
    - [x]  Every asset listed for IRM update in forum has both a payload action and a test
    - [x]  Verify the deployed IRM contract
        - [x]  Contract is verified on Etherscan
        - [x]  Source matches `lib/sparklend-advanced/src/RateTargetKinkInterestRateStrategy.sol`
        - [x]  Constructor args decoded from Etherscan match expected values:
            - [x]  `provider` , correct PoolAddressesProvider - `POOL_ADDRESSES_PROVIDER`
            - [x]  `rateSource` , correct rate oracle (same as old IRM's RATE_SOURCE)
            - [x]  `optimalUsageRatio` - unchanged
            - [x]  `baseVariableBorrowRate` (Kink) - 0 - unchanged
            - [x]  `variableRateSlope1Spread` (Kink) - `0.005e27`
            - [x]  `variableRateSlope2`  - `0.15e27` - unchanged
            - [x]  `_assertBytecodeMatches()` passes in the test (auto-verified by `_testRateTargetKinkIRMUpdate` helper)
    - [x]  Manual bytecode verification :
        
        ```solidity
        ➜  sparklend-advanced git:(master) POOL_ADDRESSES_PROVIDER="0x02C3eA4e34C0cBd694D2adFa2c690EECbC1793eE" RATE_SOURCE="0x57027B6262083E3aC3c8B2EB99f7e8005f669973" OPTIMAL_USAGE_RATIO="950000000000000000000000000" BASE_BORROW_RATE="0" SLOPE_1="5000000000000000000000000" SLOPE_2="150000000000000000000000000"
        ➜  sparklend-advanced git:(master) forge verify-bytecode 0x4E494988E68e6Fc52309BE4937869e27F0C304AC RateTargetKinkInterestRateStrategy --encoded-constructor-args $(cast abi-encode "constructor(address,address,uint256,uint256,int256,uint256)" $POOL_ADDRESSES_PROVIDER $RATE_SOURCE $OPTIMAL_USAGE_RATIO $BASE_BORROW_RATE $SLOPE_1 $SLOPE_2)
        
        Verifying bytecode for contract RateTargetKinkInterestRateStrategy at address 0x4E494988E68e6Fc52309BE4937869e27F0C304AC
        [⠊] Compiling...
        No files changed, compilation skipped
        Creation code matched with status full
        Runtime code matched with status full
        ```
        
- Deprecate SparkLend reserve  - rETH - [3] - DONE
    - [x]  Freeze the reserve according to the forum post - `rETH`
    - [x]  Check that the reserve is frozen
    - [x]  Check that the user can’t supply or borrow that reserve after spell executes - `test_ETHEREUM_sparkLend_deprecateRETH_e2e`
    - [x]  Check the test before and after LTV and isFrozen status - `test_ETHEREUM_sparkLend_deprecateRETH`
- Withdraw reserves - [4] - DONE
    - [x]  Review SparkPayloadEthereum code for claiming all reserves
    - [x]  Ensure stablecoins go to ALMProxy and everything else goes to Liquidation multisig
    - [x]  Review that test decrements and increments funds from correct addresses for all reserves
- Treasury transfer - [5a] - DONE
    - [x]  Check receiver address against forum post or address registry
        - [x]  To: `SPARK_FOUNDATION_MULTISIG=0x92e4629a4510AF5819d7D1601464C233599fF5ec`
    - [x]  Check amount against forum post
        - [x]  Amount: `1_100_000e18` USDS
    - [x]  Review that test decrements and increments funds from correct addresses
- Treasury transfer - [5b] - DONE
    - [x]  Check receiver address against forum post or address registry
        - [x]  To: `SPARK_ASSET_FOUNDATION_MULTISIG= 0xEabCb8C0346Ac072437362f1692706BA5768A911`
    - [x]  Check amount against forum post
        - [x]  Amount: `100_000e18` USDS
    - [x]  Review that test decrements and increments funds from correct addresses

**Comment by certora-spark-review** [comment]
Approving the payload at `0x160158d029697FEa486dF8968f3Be17a706dF0F0`

**Review by lucas-manuel — APPROVED** [review]
Approving deployed spell at `0x160158d029697FEa486dF8968f3Be17a706dF0F0`

**Comment by deluca-mike** [comment]
Approving deployed spell at `0x160158d029697FEa486dF8968f3Be17a706dF0F0`

**Review by deluca-mike — APPROVED** [review]
